cli/src/main/java/com/devonfw/tools/ide/url/updater/AbstractUrlUpdater.java

```diff
@@ -377,43 +384,50 @@ protected boolean isValidContentType(String contentType) {
    * @param urlVersion the {@link UrlVersion} where to store the collected information like status and checksum.
    * @param os the {@link OperatingSystem}
    * @param architecture the {@link SystemArchitecture}
+   * @param checksum the existing checksum (e.g. from JSON metadata) or the empty {@link String} if not available and computation needed.
    * @return {@code true} if the download was checked successfully, {@code false} otherwise.
    */
   private boolean checkDownloadUrl(String edition, String url, UrlVersion urlVersion, OperatingSystem os,
       SystemArchitecture architecture, String checksum) {
 
     HttpResponse<?> response = doCheckDownloadViaHeadRequest(url);
     int statusCode = response.statusCode();
-    String tool = getToolWithEdition(edition);
+    String toolWithEdition = getToolWithEdition(edition);
     String version = urlVersion.getName();
 
-    boolean success = isValidDownload(url, tool, version, response);
+    boolean success = isValidDownload(url, toolWithEdition, version, response);
 
-    // Checks if checksum for URL is already existing
-    UrlDownloadFile urlDownloadFile = urlVersion.getUrls(os, architecture);
-    if (urlDownloadFile != null) {
-      UrlChecksum urlChecksum = urlVersion.getChecksum(urlDownloadFile.getName());
-      if (urlChecksum != null) {
-        logger.warn("Checksum is already existing for: {}, skipping.", url);
-        doUpdateStatusJson(success, statusCode, edition, urlVersion, url, true);
-        return true;
```

Comment:
evil bug: here the `success` status was lost and always `true` was returned even if the head requested resulted in an invalid response.